Add draft for experiment approvals (#1009)
amplitude/amplitude-docs: content/collections/experiment/en/managing-flags-and-experiments-with-approvals.md

---
id: c7210fe3-77ed-43ee-9f76-7ad20736c439
blueprint: experiment
title: 'Managing flags and experiments with approvals'
this_article_will_help_you:
  - 'Learn how to use the approvals workflows for feature flags and experiments'
landing: false
exclude_from_sitemap: false
updated_by: e6d8f28f-00c6-426e-9ab9-664028ddc50c
updated_at: 1758133452
---
Improve the governance of your experimentation program and decrease the risk of unintended changes by requiring approvals for critical changes to experiment configuration.

Experiment Approvals is only available to Growth and Enterprise customers.

## Setting up approvals

Turn on approvals by going to _Organization Settings > Experiment > Approvals_. 

> Only users who have manager or admin roles can modify approvals settings.

Reviewer: @shelleywang Do you know if this will be a granular privilege with RBAC?

Add the individual projects that you would like to require approvals for.

For each project, you can specify one of the following options: 

> 1. **Peer approvals**: Any users with access can respond to pending approval requests

Reviewer: ```suggestion

1. **Peer approvals**: Any users with access can respond to pending approval requests
```

> 2. **Specific approvers**: Only the users designated will be able to respond to pending approval requests

Reviewer: ```suggestion
2. **Specific approvers**: Only the users designated will be able tocan respond to pending approval requests
```

> Additionally, admin users will always be able to respond to pending approvals.

Reviewer: ```suggestion
Additionally, admin users can respond to pending approvals.
```

## Requesting and responding to approval requests

### Requiring approvals to activate flags and experiments

> After approvals are enabled for a project, your organization will follow this approvals process: 

Reviewer: ```suggestion
After approvals are enabled for a project, Amplitude recommends that you follow this approval process: 
```

> 1. When starting or scheduling an experiment or activating or scheduling a feature flag, the requestor must select approver(s) to notify.

Reviewer: ```suggestion
1. When starting or scheduling an experiment or activating or scheduling a feature flag, the requestor selects one or more approver to notify.
```

2. The experiment displays a “Pending Approval” status until it's approved.
3. Approvers can review the experiment and either approve the requested changes or reject the changes.
4. The approval or rejection automatically notifies the original requestor of the response.

While the approval is pending, users can make additional changes to the flag or experiment configuration, enter or exit testing mode, or cancel the request at any time.

> After the approval is approved for scheduled flags and experiments, users can still modify any configuration until the scheduled start date.

Reviewer: ```suggestion
When the scheduled flag or experiment gains approval, users with access can modify any setting until the scheduled start date.
```

### Requiring approvals for critical changes to live flags and experiments

> When a flag or experiment is active, making some types of changes will also require approval:

Reviewer: ```suggestion
When a flag or experiment is active, certain updates also require approval:
```

| Field    | Type of Change    |
| --- | --- |
| Target Segments | Adding and removing segments, modifying conditions or bucketing.    |
| Variants | Any changes, including adding, renaming or removing variants.    |
| Variant Distribution | Any changes |
| Exposure Event | Any changes |
| Bucketing Salt | Any changes | 
| Sticky Bucketing | Enabling / Disabling | 

When reviewing these approval requests, users can view the full list of changes in the approval banner.

> While these approvals are pending, the flag or experiment will also be locked down and users will be unable to make any additional changes until the approval is cancelled or completed.

Reviewer: ```suggestion
When approvals are pending, the flag or experiment is locked. This prevents users with access from making other changes until the previous approval request is completed or cancelled.
```
